packages/ui/react-ui-editor/src/extensions/comments.ts

```diff
@@ -151,6 +155,65 @@ export type CommentsOptions = {
   onHover?: (el: Element, shortcut: string) => void;
 };
 
+const trackPastedComments = (onMove: (threadID: string, cursor: string) => void) => {
+  let tracked: { text: Text; comments: { from: number; to: number; id: string }[] } | null = null;
+
+  const registerCopy = (event: Event, view: EditorView) => {
+    const comments = view.state.field(commentsStateField);
+    const { main } = view.state.selection;
+    const inSel = comments.ranges.filter(
+      (range) => range.from >= main.from && range.to <= main.to && range.from < range.to,
+    );
+    if (!inSel.length) {
+      tracked = null;
+    } else {
+      tracked = {
+        text: view.state.doc.slice(main.from, main.to),
+        comments: inSel.map((range) => ({ from: range.from - main.from, to: range.to - main.from, id: range.id })),
+      };
+    }
+  };
+
+  return [
+    EditorView.domEventHandlers({
+      cut: registerCopy,
```

Comment:
How can we handle undo?
E.g., cut paragraph containing selection, then ctrl-z.
Potentially with history (i.e., multiple times).